test/integration/subprocess_mixin.py

```diff
@@ -0,0 +1,74 @@
+"""Mixin for managing child subprocess with an additional thread."""
+
+from threading import Thread, Condition
+import subprocess
+import logging
+from abc import ABC, abstractmethod
+
+
+class SubprocessMixin(ABC):
+  """Mixin used to manage launching subprocess using a separate Python thread."""
```

Comment:
Can we expand the class docstring with an example use, so it is obvious in what order are the methods supposed to be called?

Reply:
makes sense, I added a note to look at the particular tests since they provide a clear example and won't go stale as if they're broken CI will complain